compiler/qsc_frontend/src/resolve.rs

```diff
@@ -771,6 +819,24 @@ fn resolve(
         }
     }
 
+    if candidates.len() > 1 {
```

Comment:
yeah, the thinking is that we may have cases where an API is deprecated in one namespace in favor of adding an updated version with the same name to another namespace. If you have both namespaces open you get an ambiguous name error, which seems silly because the compiler knows that one of them is deprecated/unimplemented and the other is not. So instead, when we see multiple symbols in scope with the same name, we only treat it as ambiguous if they are both valid and resolve to the non-deprecated one if we can.

This scenario of moving a callable from one namespace to another and changing nothing else came up with some frequency before, and @DmitryVasilevsky already has some plans to do some clean-up with that again, so we know we are going to run into this case and wanted to avoid forcing the user to use Fully-Qualified Names everywhere.